### PR TITLE
fix tooltip disabled status evaluator

### PR DESCRIPTION
**Pull request by Fercas123**
Fixing issue with tooltip:
If formfield disabled is undefined, the disabled prop being explicitly passed into tooltip gets ignored.

### CODE REVIEW

File: .changeset/plenty-trees-laugh.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@salt-ds/core": patch
+---
+
+Fix logic for tooltip disabled to evaluate disabledProp is passed
```
Comment (origami-z) [inline .changeset/plenty-trees-laugh.md:5]:
```suggestion
Fixed Tooltip not honoring `disabled` prop when used in Form Field
```

**Review by origami-z — DISMISSED** [review]

File: packages/core/src/tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -89,7 +89,8 @@ export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
       validationStatus: formFieldValidationStatus,
     } = useFormFieldProps();
 
-    const disabled = formFieldDisabled ?? disabledProp;
+    const disabled = Boolean(formFieldDisabled) || disabledProp;
```
Comment (alycrys) [inline packages/core/src/tooltip/Tooltip.tsx:92]: Can you add a story or a test to show the bug? I tried 
`
  <FormField disabled>
    <Tooltip {...props}>
      <Button>Hover</Button>
    </Tooltip>
  </FormField>
`
and it doesn't show the tooltip.

Comment (origami-z) [reply packages/core/src/tooltip/Tooltip.tsx:92]: Is this expected? disabled tooltip won't show?

Comment (alycrys) [reply packages/core/src/tooltip/Tooltip.tsx:92]: do you need to display a tooltip even if no truncation?

File: packages/core/src/__tests__/__e2e__/tooltip/Tooltip.cy.tsx [hunk 3]
```diff
@@ -57,6 +57,16 @@ describe("GIVEN a Tooltip", () => {
     });
   });
 
+  describe.only("WHEN disabled", () => {
```
Comment (feliciackh) [inline packages/core/src/__tests__/__e2e__/tooltip/Tooltip.cy.tsx:60]:
```suggestion
  describe("WHEN disabled", () => {
```

File: packages/core/src/tooltip/Tooltip.tsx (hunk 2 above)
Comment (Fercas123) [reply packages/core/src/tooltip/Tooltip.tsx:92]: exactly

File: packages/core/src/__tests__/__e2e__/tooltip/Tooltip.cy.tsx (hunk 3 above)
Comment (Fercas123) [reply packages/core/src/__tests__/__e2e__/tooltip/Tooltip.cy.tsx:60]: nice spot!!!

File: packages/core/src/tooltip/Tooltip.tsx (hunk 2 above)
Comment (joshwooding) [reply packages/core/src/tooltip/Tooltip.tsx:92]: Ultimately the logic here is wrong, with two booleans they should be "OR"d but in the case it's using the undefined check

File: packages/core/src/__tests__/__e2e__/tooltip/Tooltip.cy.tsx [hunk 4]
```diff
@@ -57,6 +57,16 @@ describe("GIVEN a Tooltip", () => {
...
+    it("should not render when form field context disabled is undefined", () => {
+      cy.mount(
+        <FormField>
+          <Default disabled />
```
Comment (joshwooding) [inline packages/core/src/__tests__/__e2e__/tooltip/Tooltip.cy.tsx:64]: I believe this is a false positive, the Tooltip hasn't been opened yet so won't be visible.

Comment (Fercas123) [reply packages/core/src/__tests__/__e2e__/tooltip/Tooltip.cy.tsx:64]: ah true, will use an open one

**Review by alycrys — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**